internal/build/resolver/resolver.go

```diff
@@ -33,6 +33,13 @@ func (ar AggregateResolver) Resolve(ctx context.Context) (*build.Build, error) {
 	return nil, nil
 }
 
+func (ar AggregateResolver) WithResolverWhen(condition bool, resovler BuildResolverFn) AggregateResolver {
+	if condition {
+		return append(ar, resovler)
+	}
+	return ar
+}
+
```

Comment:
This is the second option, see below as well. It allows chaining to include a resolver depending on the evaluation of a condition. So we could move the inclusion of the resolver out to the command level and only use it if the repo is not nil

Reply:
I prefer this option!

Reply:
Oh good point. So we need the first option regardless really - just to make absolute sure